common/perception_utils/include/perception_utils/transform.hpp

```diff
@@ -15,71 +15,18 @@
 #ifndef PERCEPTION_UTILS__TRANSFORM_HPP_
 #define PERCEPTION_UTILS__TRANSFORM_HPP_
 
-#include "geometry_msgs/msg/transform.hpp"
-
 #include <tf2_ros/buffer.h>
 #include <tf2_ros/transform_listener.h>
 
-#ifdef ROS_DISTRO_GALACTIC
-#include <tf2_geometry_msgs/tf2_geometry_msgs.h>
-#else
-#include <tf2_geometry_msgs/tf2_geometry_msgs.hpp>
-#endif
-
-#include "boost/optional.hpp"
-
 #include <string>
 
-namespace
-{
-[[maybe_unused]] boost::optional<geometry_msgs::msg::Transform> getTransform(
-  const tf2_ros::Buffer & tf_buffer, const std::string & source_frame_id,
-  const std::string & target_frame_id, const rclcpp::Time & time)
-{
-  try {
-    geometry_msgs::msg::TransformStamped self_transform_stamped;
-    self_transform_stamped = tf_buffer.lookupTransform(
-      target_frame_id, source_frame_id, time, rclcpp::Duration::from_seconds(0.5));
-    return self_transform_stamped.transform;
-  } catch (tf2::TransformException & ex) {
-    RCLCPP_WARN_STREAM(rclcpp::get_logger("perception_utils"), ex.what());
-    return boost::none;
-  }
-}
-}  // namespace
-
 namespace perception_utils
 {
+
 template <class T>
 bool transformObjects(
   const T & input_msg, const std::string & target_frame_id, const tf2_ros::Buffer & tf_buffer,
-  T & output_msg)
-{
-  output_msg = input_msg;
+  T & output_msg);
```

Comment:
Ah, I think you can't write template functions in `.cpp`.
If you do so, you have to instantiate the types you use.

Reply:
Sorry, I miss it.
Fixed at [d7f668c](https://github.com/autowarefoundation/autoware.universe/pull/2028/commits/d7f668c0459506d35409eb4edd26f0dabfde62ac)

Reply:
By consulting with @kenji-miyake, change to detail namespace.